src/app/messages-page/messages-page.component.ts

```diff
@@ -1,118 +1,264 @@
+      });
+
+      this.publicKeyToProfileMap[item.PublicKeyBase58Check] = item;
+
+      const TimestampNanos = Date.now() * 1e6;
+      const threadListItem: DecryptedMessageEntryResponse = {
+        ChatType: ChatType.DM,
+        SenderInfo: {
+          OwnerPublicKeyBase58Check: SenderPublicKeyBase58Check,
+          AccessGroupKeyName: SenderAccessGroupKeyName,
+          AccessGroupPublicKeyBase58Check: SenderAccessGroupPublicKeyBase58Check,
+        },
+        RecipientInfo: {
+          OwnerPublicKeyBase58Check: RecipientPublicKeyBase58Check,
+          AccessGroupKeyName: RecipientAccessGroupKeyName,
+          AccessGroupPublicKeyBase58Check: RecipientAccessGroupPublicKeyBase58Check,
+        },
+        MessageInfo: {
+          EncryptedText: "",
+          TimestampNanos,
+          TimestampNanosString: TimestampNanos.toString(),
+          ExtraData: {},
+        },
+        DecryptedMessage: "",
+        IsSender: true,
+        error: "",
+      };
+      this.threadPreviewList.unshift(threadListItem);
+      this.selectThread(threadListItem);
+    } catch (e) {
+      this.globalVars._alertError(e?.error?.error ?? e?.message);
+    }
   };
 
   constructor(
     public globalVars: GlobalVarsService,
-    private backendApi: BackendApiService,
-    private router: Router,
-    private titleService: Title,
     private modalService: BsModalService,
-    private tracking: TrackingService
+    private route: ActivatedRoute,
+    private router: Router,
+    private backendApi: BackendApiService
   ) {}
 
   ngOnInit() {
-    this.titleService.setTitle(`Messages - ${environment.node.name}`);
-  }
+    if (this.globalVars.loggedInUser) {
+      this.updateThreadList().then(() => {
+        this.route.queryParams.subscribe((queryParams) => {
+          const firstThread = this.threadPreviewList?.[0];
 
-  openNewMessageModal() {
-    const modalClass = this.globalVars.isMobile()
-      ? "modal-dialog-centered modal-dialog-high modal-dialog-light"
-      : "modal-dialog-centered";
-    const backdrop = !this.globalVars.isMobile();
-    const messageSelectorModal = this.modalService.show(MessageRecipientModalComponent, {
-      class: modalClass,
-      backdrop,
-      animated: !this.globalVars.isMobile(),
-    });
-    messageSelectorModal.content.userSelected.subscribe((event) => {
-      this.messagesInboxComponent._handleCreatorSelectedInSearch(event);
-    });
+          if (queryParams.username) {
+            this.backendApi.GetSingleProfile("", queryParams.username).subscribe((res) => {
+              const selectProfileAttempt = res.Profile ? this.onSearchItemSelected(res.Profile) : Promise.resolve();
+              selectProfileAttempt.then(() => {
+                this.router.navigate([], {
+                  queryParams: { username: null },
+                  queryParamsHandling: "merge",
+                  replaceUrl: true,
+                });
+              });
+            });
+          } else {
```

Comment:
a custom logic to handle query params like `/messages?username=superzordon` that should create/jump to the conversation with the specified user

Reply:
Sweet 🔥